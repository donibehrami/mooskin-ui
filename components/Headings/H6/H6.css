@import url('https://fonts.googleapis.com/css?family=Convergence');

.h6 {
    font-family: 'Convergence', sans-serif;
    font-size: 0.625rem;
    font-weight: 400;
    color: #3a4257;
}