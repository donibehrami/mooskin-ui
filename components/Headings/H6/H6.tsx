import * as React from 'react';

import styles from './H6.css';

import {IHeadingProps} from '../props';

class H6 extends React.Component<IHeadingProps, {}> {

    static defaultProps = {
        className: '',
        style: {}
    };

    static displayName = 'H6';

    render() {

        const {id, className, style, children} = this.props;

        const classes = `h6-component ${styles.h6} ${className}`;

        return (
            <div
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </div>
        );
    }
}

export default H6;
