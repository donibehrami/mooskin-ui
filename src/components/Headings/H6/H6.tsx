import * as React from 'react';

import styles from './H6.css';

import {IHeaderProps} from '../../../types/commonTypes';

class H6 extends React.Component<IHeaderProps, void> {

    public static defaultProps = {
        className: '',
        style: {}
    };

    public render() {

        const {id, className, style, children} = this.props;

        const classes = `button-component ${styles.h6} ${className}`;

        return (
            <h6
                id={id}
                className={classes}
                style={style}
            >
                {children}
            </h6>
        );
    }
}

export default H6;
