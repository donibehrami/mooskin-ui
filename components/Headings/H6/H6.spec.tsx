import * as React from 'react';
import H6 from './H6';

import { shallow } from 'enzyme';

describe('H6', () => {

    test('renders correctly', () => {

        const tree = shallow(
            <H6
                className="myClass"
                style={{color: 'blue'}}
                id={'heading6'}
            >
                Mooskin
            </H6>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders properly into dom with text, id and class', () => {

        const component = shallow(<H6 id="Campaigns" className="headers">Mooo!</H6>);

        expect(component.find('p').text()).toBe('Mooo!');
        expect(component.find('p').prop('id')).toEqual('Campaigns');
        expect(component.find('p').hasClass('headers')).toBe(true);
    });

});
