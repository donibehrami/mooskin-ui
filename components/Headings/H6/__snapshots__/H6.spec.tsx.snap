// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H6 renders correctly 1`] = `
<p
  className="h6-component h6 myClass"
  id="heading6"
  style={
    Object {
      "color": "blue",
    }
  }
>
  Mooskin
</p>
`;
